### ISSUE TITLE
Portability of softwares that belong to a raysession

### ISSUE THREAD

**Issue by newlaurent62** [Houston4444/RaySession#72]
Hello,

I have some programs in my raysession document that have been compiled from their master branch and are not yet available as a deb package in official repositories ...

I place them in a session folder and create a link in the proxy dirs but there is still the problem of the icon.

I would like to tell RaySession that icon, specific to this raysession document have been added.

for example, i would like to add two icons for Jamulus software compiled from source, one for client mode and another for server mode.

They could be in proxy-dir/share/icons or in a session-dir/.local/share/icons ?

BR,
Laurent

**Comment by Houston4444** [comment]
> I place them in a session folder and create a link in the proxy dirs but there is still the problem of the icon.

You mean, you place a program code source in raysession path ? that can works but it's not designed to work this way. It is designed to work with installed softwares.
I don't manage icon paths, system does.
You can use absolute path (I think) to specify icons. The problem is that, if this is in the session, it will not works anymore when session has been duplicated.

This is quite a niche use case, I don't want to take time to implement this but I could accept a PR to search icons in session-dir/.local/share/icons. It should be in src/gui/list_widget_clients.py (also in child_dialogs.py for client properties dialog, but I moved this class in so_boring_proxy).

**Comment by newlaurent62** [comment]
> You mean, you place a program code source in raysession path ? that can works but it's not designed to work this way. It is designed to work with installed softwares.

No, i only copied the compiled program to the session dir.

> This is quite a niche use case, I don't want to take time to implement this but I could accept a PR to search icons in session-dir/.local/share/icons. It should be in src/gui/list_widget_clients.py (also in child_dialogs.py for client properties dialog, but I moved this class in so_boring_proxy branch).

Thanks for the information, I will have a look

**Comment by newlaurent62** [comment]
On master branch, icons in $session-dir/.local/share/icons/32x32 cannot be found.

Do you plan to implement this ?

**Comment by Houston4444** [comment]
Not for session because if a client is set as template, template will never found the icon.

I think to something like that : a folder named ray-data.$CLIENT_ID, in that place we could add search folders.

ray-data.$CLIENT_ID
- share/
- share/pixmaps/
- share/icons/
- share/applications/
- bin/

and more if we have more idea.

**Comment by newlaurent62** [comment]
> ray-data.$CLIENT_ID

    share/
    share/pixmaps/
    share/icons/
    share/applications/
    bin/


Why not in $session_path/$session_name.$clientID/

**Comment by newlaurent62** [comment]
Yes i understand, the name of the session cannot be in the template.

ray-app/$CLIENT_ID wouldn't be more accurate ?

**Comment by newlaurent62** [comment]
$session_name/$CLIENT_ID already contains "data" (audio files, config files ...)

**Comment by Houston4444** [comment]
$SESSION_NAME.$CLIENT_ID is the default path but it can also be $CLIENT_NAME.$CLIENT_ID or $CUSTOM_PREFIX.$CLIENT_ID depending on how is set prefix_mode and prefix.

The problem with this directory is that we don't know what client could put inside, so for security, a parallel dir is better.

**Comment by newlaurent62** [comment]
From my user point of view, it would be best to have one repository per
session rather than client application.

In my session template, I can have several clients related to one
application with different arguments.

If later we decide to put appimage or whatever in the session we know we
have a local repository to do so.

Adding client dirs will introduce confusing elements unless they are hidden
files.

Le mer. 24 juin 2020 à 17:34, Houston4444 <notifications@github.com> a
écrit :
